pytensor/tensor/rewriting/linalg.py

```diff
@@ -539,3 +540,90 @@ def svd_uv_merge(fgraph, node):
                     or len(fgraph.clients[cl.outputs[2]]) > 0
                 ):
                     return [cl.outputs[1]]
+
+
+def _find_solve_with_eye(node) -> bool:
+    """
+    The result of solve(A, b) is the solution x to the linear equation Ax = b. If b is an identity matrix (Eye), x is simply inv(A).
+    Here, we are just recognising whether the solve operation returns an inverse or not; not replacing it because solve is mathematically more stable than inv.
+    """
+    valid_solves = (Solve, SolveTriangular)
+    # First, we verify whether we have a valid solve op
+    if not (
+        isinstance(node.op, Blockwise) and isinstance(node.op.core_op, valid_solves)
+    ):
+        return False
+    # If the current op is solve, we check for b. If b is an identity matrix (Eye), we can return True
+    solve_inputs = node.inputs
+    eye_input = solve_inputs[1].owner
+
+    # We check for b = Eye and also make sure that if it was an Eye, then k = 0 (1's are only across the main diagonal)
+    if not (eye_input and isinstance(eye_input.op, Eye)):
+        return False
+
+    if eye_input.inputs[-1].data.item() != 0:
```

Comment:
This is assuming eye_input.inputs[-1] is a Constant which is not guaranteed. You first have to check for that